packages/ElasticSearchDSL/src/Rector/MethodCall/MigrateFilterToQueryRector.php

```diff
@@ -0,0 +1,84 @@
+<?php declare(strict_types=1);
+
+namespace Rector\ElasticSearchDSL\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\NodeDumper;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+/**
+ * @see https://github.com/ongr-io/ElasticsearchDSL/blob/5.x/CHANGELOG.md
+ */
+final class MigrateFilterToQueryRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Migrates addFilter to addQuery', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        $search = new \ONGR\ElasticsearchDSL\Search();
+
+        $search->addFilter(
+            new \ONGR\ElasticsearchDSL\Query\TermsQuery('categoryIds', [1, 2])
+        );
+    }
+}
+CODE_SAMPLE
+,
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        $search = new \ONGR\ElasticsearchDSL\Search();
+
+        $search->addQuery(
+            new \ONGR\ElasticsearchDSL\Query\TermsQuery('categoryIds', [1, 2]),
+            \ONGR\ElasticsearchDSL\Query\Compound\BoolQuery::FILTER
+        );
+    }
+}
+CODE_SAMPLE
+
+            )
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isType($node, 'ONGR\ElasticsearchDSL\Search')) {
+            return null;
+        }
+
+        if (! $node->name instanceof Node\Identifier) {
+            return null;
+        }
+
+        if ($node->name->name !== 'addFilter') {
```

Comment:
better `isName()`

```suggestion
        if (! $this->isName($node, 'addFilter')) {
```

And remove the part above:

```diff
-if (! $node->name instanceof Node\Identifier) {
-    return null;
-}
```